packages/react-composites/src/composites/common/Composite.tsx

```diff
@@ -23,9 +23,15 @@ export interface BaseCompositeProps {
   /**
    * Locale for the composite.
    *
-   * @defaultValue English (US)
+   * @defaultValue English (US) locale
    */
-  locale?: CompositeLocale;
+  locale?: CompositeLocale | ((locale?: string) => Promise<CompositeLocale>);
+  /**
+   * Default string for locale prop if locale prop is type (locale?: string) => Promise<CompositeLocale>
+   *
+   * @defaultValue undefined
+   */
+  defaultLocale?: string;
```

Comment:
I am confused  by how the async locale loading works. What provides the non-default locale string to be used for calling the locale loading function?

Reply:
Resolved since we've dropped the argument to loader now.